### ISSUE TITLE
OnionShare doesn't quit when closing the window in specific circumstance

### ISSUE THREAD

**Issue by micahflee** [onionshare/onionshare#1566]
Most of the time when I close the OnionShare window it quits the process correctly, but sometimes it doesn't. I seem to be able to reproduce it by:

- Open OnionShare to the autoconnect tab
- Click Connect to Tor, then cancel
- Click Try Bridge, then cancel
- Click Try Bridge, then cancel again
- Close the window

In this case, the window closes but the process stays open. The verbose logs look like:

```
[Mar 28 2022 19:14:50] CensorshipCircumvention.__init__
[Mar 28 2022 19:14:50] CensorshipCircumvention.__init__: Using Meek with CensorshipCircumvention API
[Mar 28 2022 19:14:52] CensorshipCircumvention.censorship_obtain_settings: No settings found for this country
[Mar 28 2022 19:14:52] Meek.cleanup
[Mar 28 2022 19:14:52] AutoConnectTab._got_no_bridges: Could not obtain bridges, so falling back to trying built-in obfs4 bridges
[Mar 28 2022 19:14:52] Settings.save
[Mar 28 2022 19:14:52] Settings.save: Settings saved in C:\Users\micah\AppData\Roaming\OnionShare\onionshare.json
[Mar 28 2022 19:14:52] AutoConnectTab._got_bridges: Got bridges. Trying to reconnect to Tor
[Mar 28 2022 19:14:52] TorConnectionWidget.start
[Mar 28 2022 19:14:52] TorConnectionThread.__init__
[Mar 28 2022 19:14:52] TorConnectionThread.run
[Mar 28 2022 19:14:52] Onion.connect: connection_type=bundled
[Mar 28 2022 19:14:52] Onion.connect: tor_data_directory_name=C:\Users\micah\AppData\Roaming\OnionShare\tor_data
[Mar 28 2022 19:14:52] Common.get_resource_path: filename=torrc_template, path=C:\Users\micah\code\onionshare\cli\onionshare_cli\resources\torrc_template
[Mar 28 2022 19:14:53] Onion.connect: Writing torrc template file
[Mar 28 2022 19:14:53] Common.get_resource_path: filename=torrc_template-obfs4, path=C:\Users\micah\code\onionshare\cli\onionshare_cli\resources\torrc_template-obfs4
[Mar 28 2022 19:14:53] Onion.connect: Wrote in the built-in bridges from torrc templates
[Mar 28 2022 19:14:53] Onion.connect: starting C:\Users\micah\code\onionshare\desktop\onionshare\resources\tor\Tor\tor.exe subprocess
[Mar 28 2022 19:14:53] Onion.connect: tor pid: 7948
[Mar 28 2022 19:14:55] Onion.connect: authenticating to tor controller
Connecting to the Tor network: 1% - Connecting to pluggable transport[Mar 28 2022 19:15:04] MainWindow.closeEvent
```

And it's just stalled. I have to press Ctrl-C to quit.

**Comment by SaptakS** [comment]
This also raises one question in my mind. When we click cancel in autoconnect tab, should it go to the bridge tab? Or should it just cancel and stay in autoconnect tab?

**Comment by mig5** [comment]
I think it should be OK to stay in autoconnect tab. That's been useful already to 'simulate' a connection problem (by cancelling the connection) in order to test the autoconnect flows that surface as a result

**Comment by SaptakS** [comment]
@micahflee I am not sure if this fixed by itself, but I am no longer able to reproduce it in linux and mac. We did change the UI a bit, and updated the tor binaries. Don't know if that somehow fixed this, or I am doing something wrong in the reproducing steps.

**Comment by micahflee** [comment]
Hmm, I also can't reproduce it anymore. And I merged #1582 so these steps don't work anymore. I'll close this issue.
